src/slam_toolbox_common.cpp

```diff
@@ -240,6 +241,8 @@ SlamToolbox::on_shutdown(const rclcpp_lifecycle::State & state)
 SlamToolbox::~SlamToolbox()
 /*****************************************************************************/
 {
+  stop_threads_.store(true);
```

Comment:
Needs to be in the `on_deactivate` function as well

Reply:
@GoesM just one thing to finish this up!

Reply:
I've added it .